docs/how-to-guides/build-directory.rst

```diff
@@ -0,0 +1,14 @@
+.. _how-to-guides-build-directory:
+
+*******************************
+Use a permanent build directory
+*******************************
+
+
+.. todo::
+
+   - Explain why you want to use the ``builddir`` option
```

Comment:
This includes things that are hard to pass through a front end's config-settings I think. Would be good to point out that this works as expected when you, for example, run `meson setup --cross-file ...`, and `meson-python` will reconfigure only what's needed.

Reply:
Could you elaborate? I don't think I follow :sweat_smile:

Reply:
I've come across this multiple times already, for cross-compilation in particular. See for example https://github.com/scipy/scipy/blob/1c836efe5ff37ffa4490756269b060a464690e62/.github/workflows/wheels.yml#L180-L188.

Also for conda-forge, you need in `build.sh` something like this for cross-compiling (the first line is how the conda-forge build tooling configures automatically for cross-compilation):
```
meson ${MESON_ARGS} builddir
python -m build -Cbuilddir=builddir
```